sdk/pinocchio/src/sysvars/slot_hashes/test.rs

```diff
@@ -0,0 +1,497 @@
+use super::test_utils::*;
+use crate::{
+    account_info::{Account, AccountInfo},
+    program_error::ProgramError,
+    sysvars::{clock::Slot, slot_hashes::*},
+};
+use core::{
+    mem::{align_of, size_of},
+    ptr,
+};
+
+extern crate std;
+use std::io::Write;
+use std::vec::Vec;
+
+#[test]
+fn test_layout_constants() {
+    assert_eq!(NUM_ENTRIES_SIZE, size_of::<u64>());
+    assert_eq!(SLOT_SIZE, size_of::<u64>());
+    assert_eq!(HASH_BYTES, 32);
+    assert_eq!(ENTRY_SIZE, size_of::<u64>() + 32);
+    assert_eq!(MAX_SIZE, 20_488);
+    assert_eq!(size_of::<SlotHashEntry>(), ENTRY_SIZE);
+    assert_eq!(align_of::<SlotHashEntry>(), align_of::<[u8; 8]>());
+    assert_eq!(
+        SLOTHASHES_ID,
+        [
+            6, 167, 213, 23, 25, 47, 10, 175, 198, 242, 101, 227, 251, 119, 204, 122, 218, 130,
+            197, 41, 208, 190, 59, 19, 110, 45, 0, 85, 32, 0, 0, 0,
+        ]
+    );
+
+    pub fn check_base58(input_bytes: &[u8], expected_b58: &str) {
+        match bs58::decode(expected_b58).into_vec() {
```

Comment:
If you update to use `five8_const`:
```suggestion
        match five8_const::decode_32_const(expected_b58).into_vec() {
```